### PR TITLE
HDDS-4198. Compile Ozone with multiple Java versions

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Compile Ozone with both Java 8 and 11 in CI.

Note: this is based on top of #1386.

https://issues.apache.org/jira/browse/HDDS-4198

## How was this patch tested?

https://github.com/adoroszlai/hadoop-ozone/runs/1063860191#step:6:10
https://github.com/adoroszlai/hadoop-ozone/runs/1063860210#step:6:10

### CODE REVIEW

**Review by vivekratnavel — APPROVED** [review]
+1 LGTM

File: .github/workflows/post-commit.yml [hunk 1]
```diff
@@ -34,10 +38,17 @@ jobs:
           key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
           restore-keys: |
             ${{ runner.os }}-pnpm-
-      - name: Execute tests
-        uses: ./.github/buildenv
+      - name: Cache for maven dependencies
+        uses: actions/cache@v2
         with:
-          args: ./hadoop-ozone/dev-support/checks/build.sh
```
Comment (elek) [inline .github/workflows/post-commit.yml:40]: We can consider doing the same with all the other steps.

**Review by elek — APPROVED** [review]
+1, thanks the patch @adoroszlai 

Good to be sure that Ozone can be compiled with JDK 11.

**Comment by adoroszlai** [comment]
Thanks @vivekratnavel for the review, and @elek for reviewing and committing it.

**Pull request merged**